### PR TITLE
feat: SOF-978 add `ConfigManifestEntrySelectFromColumn`

### PR DESCRIPTION
**Pull request by ianshade**
Adds new config manifest entry type, called `ConfigManifestEntrySelectFromColumn`, which allows populating dropdowns (single or multiple choice) with all values from a column in a different (or the same) config table. 
for Show Style Variants, if the variant does not override the source table, values from the Base Show Style are used. 

### An example (with a multiselect):
For a table with the following text values,
image
the following dropdowns can be automatically populated in another table (or as a single config field)
image
If a value that was previously selected no longer exists in the source column, the dropdown will have a red border, and the option will be crossed out
image

### CODE REVIEW

**Comment by Julusian** [comment]
> If a value that was previously selected no longer exists in the source column, the dropdown will have a red border, and the option will be crossed out

I assume this applies to other places that dropdowns are used too?

**Comment by ianshade** [comment]
@Julusian correct. Is that something NRK would want or not? I know some fields that fail validation have solid red background (e.g. JSON). 
And this also sort of fixes a bug that makes it impossible to uncheck a value if it no longer exists in the options, for example in the Studio where you can select compatible Show Styles, this is what used to happen when you removed a Show Style:
image
and after this change (items on the screenshots don't match because they were taken on different installations):
image

File: meteor/client/lib/EditAttribute.tsx [hunk 1]
```diff
@@ -539,15 +539,18 @@ const EditAttributeDropdown = wrapEditAttribute(
...
+			const option = _.find(this.getOptions().options, (o) => {
 				return o.value + '' === event.target.value + ''
 			})
 
 			const value = option ? option.value : event.target.value
 
 			this.handleUpdate(this.props.optionsAreNumbers ? parseInt(value, 10) : value)
 		}
-		getOptions(addOptionForCurrentValue?: boolean) {
+		getOptions(addOptionForCurrentValue?: boolean): {
```
Comment (LindvedKrvang) [inline meteor/client/lib/EditAttribute.tsx:550]: Would it be an idea to make an interface for this return type?

**Review by LindvedKrvang — APPROVED** [review]

File: meteor/client/lib/EditAttribute.tsx [hunk 2]
```diff
@@ -600,14 +603,15 @@ const EditAttributeDropdown = wrapEditAttribute(
 				}
 			}
 
+			const currentValue = this.getAttribute()
+			const currentOption = _.find(options, (o) => {
+				if (Array.isArray(o.value)) {
+					return _.contains(o.value, currentValue)
+				}
+				return o.value === currentValue
+			})
```
Comment (KvelaGorrrrnio) [inline meteor/client/lib/EditAttribute.tsx:612]: Just a suggestion.

```suggestion
			const currentOption = _.find(options, (o) => 
				Array.isArray(o.value) ? _.contains(o.value, currentValue) : o.value === currentValue
			)
```

**Review by KvelaGorrrrnio — APPROVED** [review]
Looks good 🚀

**Comment by ianshade** [comment]
@Julusian yes, it is only a visual hint

**Pull request merged**